packages/driver/test/cypress/integration/issues/5274_spec.coffee

```diff
@@ -0,0 +1,88 @@
+
+    cy.on "fail", (err) ->
+      expect(err.message).to.contain "cy.request() failed trying to load:"
+      expect(err.message).to.not.contain "ERR_UNESCAPED_CHARACTERS"
+      done()
+
+  it "should request url with ’ escaped in pathname", (done) ->
+    cy.request(encodeURI('http://localhost:1234/’'))
+
+    cy.on "fail", (err) ->
+      expect(err.message).to.contain "cy.request() failed trying to load:"
+      expect(err.message).to.not.contain "ERR_UNESCAPED_CHARACTERS"
+      done()
+
+  it "should visit url with Unicode in pathname from BMP to Astral Plane", (done) ->
+    cy.on "fail", (err) ->
+      expect(err.message).to.contain "cy.request() failed trying to load:"
+      expect(err.message).to.not.contain "ERR_UNESCAPED_CHARACTERS"
+
+    cy.request('http://localhost:1234/%')
+    cy.request('http://localhost:1234/’')
+    cy.request('http://localhost:1234/£')
+    cy.request('http://localhost:1234/Ȥ')
+    cy.request('http://localhost:1234/˵')
+    cy.request('http://localhost:1234/֍')
+    cy.request('http://localhost:1234/ץ')
+    cy.request('http://localhost:1234/ص')
+    cy.request('http://localhost:1234/ޥ')
+    cy.request('http://localhost:1234/ࠊ')
+    cy.request('http://localhost:1234/ࢨ')
+    cy.request('http://localhost:1234/⍸')
+    cy.request('http://localhost:1234/㇇')
+    cy.request('http://localhost:1234/ヸ')
+    cy.request('http://localhost:1234/ㇻ')
+    cy.request('http://localhost:1234/𓌶')
+    cy.request('http://localhost:1234/🜈')
+    cy.request('http://localhost:1234/🠋')
+    cy.request('http://localhost:1234/👩')
+    cy.request('http://localhost:1234/😀')
+    done()
+
+  it "should request url with any Unicode escaped character in pathname", (done) ->
+    cy.on "fail", (err) ->
+      expect(err.message).to.contain "cy.request() failed trying to load:"
+      expect(err.message).to.not.contain "ERR_UNESCAPED_CHARACTERS"
+
+    cy.request(encodeURI('http://localhost:1234/%'))
+    cy.request(encodeURI('http://localhost:1234/’'))
+    cy.request(encodeURI('http://localhost:1234/£'))
+    cy.request(encodeURI('http://localhost:1234/Ȥ'))
+    cy.request(encodeURI('http://localhost:1234/˵'))
+    cy.request(encodeURI('http://localhost:1234/֍'))
+    cy.request(encodeURI('http://localhost:1234/ץ'))
+    cy.request(encodeURI('http://localhost:1234/ص'))
+    cy.request(encodeURI('http://localhost:1234/ޥ'))
+    cy.request(encodeURI('http://localhost:1234/ࠊ'))
+    cy.request(encodeURI('http://localhost:1234/ࢨ'))
+    cy.request(encodeURI('http://localhost:1234/⍸'))
+    cy.request(encodeURI('http://localhost:1234/㇇'))
+    cy.request(encodeURI('http://localhost:1234/ヸ'))
+    cy.request(encodeURI('http://localhost:1234/ㇻ'))
+    cy.request(encodeURI('http://localhost:1234/𓌶'))
+    cy.request(encodeURI('http://localhost:1234/🜈'))
+    cy.request(encodeURI('http://localhost:1234/🠋'))
+    cy.request(encodeURI('http://localhost:1234/👩'))
+    cy.request(encodeURI('http://localhost:1234/😀'))
+    done()
```

Comment:
calling `done` here will cause the test to immediately pass and it not actually do anything.

Reply:
I've refactored the tests cases to reduce the test with one request by test.

Also, now, the tests are separated into to parts, one who check that there is no 'UNESCAPED_CHARACTER' error, and another one who effectively check if the result value of the url is the one we expect (properly encoded).